sshd-common/src/main/java/org/apache/sshd/common/util/buffer/Buffer.java

```diff
@@ -341,6 +341,39 @@ public String getString() {
         return GenericUtils.isEmpty(values) ? Collections.emptyList() : Arrays.asList(values);
     }
 
+    public List<OpenSshCertificate.CertificateOption> getCertificateOptions() {
+        return getCertificateOptions(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * According to <A HREF=
+     * "https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.certkeys#L222-L262">PROTOCOL.certkeys</A>:
+     *
+     * Critical Options is a set of bytes that is
+     *
+     * [overall length][name(string)][data(string)]...
+     *
+     * Where each Critical Option is encoded as a name (string) and data (string)
+     *
+     * Then the entire name + data strings are added as bytes (which will get a length prefix)
+     *
+     * @return The parsed result
```

Comment:
Please document here the null return when there are no options.

Also missing @param in the javadoc.